core/account-id/src/lib.rs

```diff
@@ -142,17 +166,22 @@ impl AccountId {
     /// use near_account_id::AccountId;
     ///
     /// let alice: AccountId = "alice.near".parse().unwrap();
-    /// assert!(!alice.is_implicit());
+    /// assert!(!alice.is_near_implicit());
     ///
     /// let rando = "98793cd91a3f870fb126f66285808c7e094afcfc4eda8a970f6648cdf0dbd6de"
     ///     .parse::<AccountId>()
     ///     .unwrap();
-    /// assert!(rando.is_implicit());
+    /// assert!(rando.is_near_implicit());
     /// ```
-    pub fn is_implicit(&self) -> bool {
+    pub fn is_near_implicit(&self) -> bool {
```

Comment:
Rather than having is_near_implicit and is_eth_implicit methods, please consider a get_account_type() method that returns and AccountType enum with values (NamedAccount, NearImplicitAccount, EthImplicitAccount). This way the compiler will force you and future developers to correctly handle all cases. It is also more extensible to any future potential account types.

Reply:
Thank you, amazing idea!

Reply:
Although there is a benefit to have a `is_implicit` method, so that we do not have to duplicate
```
let is_implicit = match account_id.get_account_type() {
    AccountType::NearImplicitAccount | AccountType::EthImplicitAccount => true,
    AccountType::NamedAccount => false,
};
``` 
in several places.

Reply:
How about adding is_implicit on the AccountType enum itself? It seems like a nice compromise where developers would still need to be forced to think about different account types but convienient enough to use.

Reply:
Sounds good! Will check it out.